docs/manifest.json

```diff
@@ -142,10 +142,6 @@
             }
           ]
         },
-        {
-          "title": "Middleware",
-          "path": "/docs/middleware.md"
```

Comment:
Looks like we want to keep this entry and update the path to `/docs/advanced-features/middleware.md` to go along with the redirect. This should resolve the failing check here https://github.com/vercel/next.js/runs/5888873149?check_suite_focus=true#step:4:5

Reply:
@ijjk Ahh, thank you! I was getting a little confused around how redirects work for these docs. So I understand for the future when we want to move something:
- Keep the location but update the path to where we want it? (Will this make it confusing for someone coming to the file in the future if this file doesn't mimic the site nav?)
- The path and redirect path should be the same, but that path uses the .md file type?

Reply:
So for non-redirect paths it should have the `.md` extension and points to a file in the docs folder, this is what the lint step is mainly checking for to ensure we don't forget to add manifest entries or have typos. 

For redirect paths it doesn't need to have `.md` as it's matching the path directly without checking the filesystem and the lint step doesn't check these.  The destination for the redirect can be any url so if we need to move docs to another site we can e.g. `https://vercel.com/docs/some/path` but in this case it can just be `/docs/advanced-features/middleware` since it's the same site. 

Seems like it would be good to add this to our contributing, I'll create a PR with notes on this and request your review.

Reply:
Opened PR with the above info here https://github.com/vercel/next.js/pull/36020

Reply:
@ijjk Thank you so much for taking the time to explain this. The added example in the contributing doc was super helpful!